### ISSUE TITLE
Cover Elasticsearch

### ISSUE THREAD

**Issue by jlevy** [open-guides/og-aws#64]
AWS' own Elasticsearch service is growing in popularity. Discuss basics, when to manage yourself vs with AWS' managed service, etc.

**Comment by jlevy** [comment]
@nitingithub want to take this one?

**Comment by zoellner** [comment]
This would probably need some comment on how AWS is lacking behind quite a bit regarding the available ES versions and the implications to get a system running securely in a VPC

**Comment by jlevy** [comment]
@zoellner yeah, what I've heard too. That's exactly the sort of tip we should put in!

**Comment by jlevy** [comment]
PR welcome, or if you'd like to put that tip here (any links/details to add?) we can incorporate it.

**Comment by dcosson** [comment]
In addition to version lagging (which right now is not actually too bad, they bumped up to 2.3 I believe in July), they have scripts disabled. And there are some relatively straightforward uses that require scripts in elasticsearch, e.g. sorting an aggregation by query score is one I just ran into.

**Comment by kimschmidtsbrain** [comment]
I can help here. I think Elasticsearch Service is one of the most powerful yet unknown service...as long as I ramp up my GitHub game lol ;-)

**Comment by jlevy** [comment]
dorrin [8:22 AM]  
I'd like to hear stories from people who switched from managed ElasticSearch to building and maintaining their own cluster

bgdnlp [8:56 AM]  
for what is worth, I think the "pets vs cattle" thing is a broad generalisation that ignores too many issues. in fact, I also think that it is dangerous from a security perspective.

rjhintz [10:01 AM]  
@dorrin Regarding: Elasticsearch. This was a recent topic on Reddit/aws. It seemed to draw some informed opinion (as opposed to "normal" Reddit.) https://www.reddit.com/r/aws/comments/55dpn6/elasticsearch_on_aws_managed_or_roll_your_own/

**Comment by DorrinPk** [comment]
Price can also be a factor. ElasticSearch service's instances are more expensive than regular EC2 instances.

**Comment by untergeek** [comment]
Piping in from the field here. AWS ES does not allow use of the /_cluster/state/metadata endpoint. This causes problems with supporting tools, like Elasticsearch Curator. 

See https://github.com/elastic/curator/issues/717

**Comment by DorrinPk** [comment]
Troubleshooting the ES service is also a problem. You can't see the instances behind the ES cluster so if you can't fix an issue by adding more memory or storage then you should contact support.

**Comment by DorrinPk** [comment]
AWS has a list of ES service limits as well : http://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/aes-limits.html Mainly no scripting, plugins, heap size limit, max instance count.

**Comment by jlevy** [comment]
Another major security gotcha to list:
http://www.zdnet.com/article/first-came-mass-mongodb-ransacking-now-copycat-ransoms-hit-elasticsearch/
